### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by ireland-advisor**
## Summary
* add oauth 2.0 for user to login through okta with okta html
* add unit/API test on users app, but haven't add test for oauth app
* add pipenv to manage the python environment, please consult "Other information" for usage
## Other information
To run this  repo successfully, you have to:

launch the environment:

`pipenv shell`

migrate the database:

`python3 manage.py migrate `

run the server:

`python3 manage.py runserver`

### CODE REVIEW

File: advisor_api/settings.py [hunk 1]
```diff
@@ -25,8 +26,15 @@
...
 DEBUG = True
 
-ALLOWED_HOSTS = ['advisor-api.herokuapp.com']
+ALLOWED_HOSTS = ['advisor-api.herokuapp.com', '127.0.0.1', 'localhost']
 
+# Okta Configuration
+ORG_URL = "https://dev-192131.okta.com"
+ISSUER = "https://dev-192131.okta.com/oauth2/default"
+CLIENT_ID = "0oa967awncYBzmmsG4x6"
+CLIENT_SECRET = "FxrXdp5YBUdPdp3wyaE3Z6NMGMBGXGJN9S3UL1CU"
```
Comment (gonac) [inline advisor_api/settings.py]: These(CLIENT_ID and CLIENT_SECRET  ) are sensitive values and the need to be kept secret

File: advisor_api/settings.py [hunk 2]
```diff
@@ -25,8 +26,15 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
```
Comment (gonac) [inline advisor_api/settings.py]: These should be false

File: advisor_api/settings.py [hunk 3]
```diff
@@ -25,8 +26,15 @@
...
-ALLOWED_HOSTS = ['advisor-api.herokuapp.com']
+ALLOWED_HOSTS = ['advisor-api.herokuapp.com', '127.0.0.1', 'localhost']
 
+# Okta Configuration
+ORG_URL = "https://dev-192131.okta.com"
+ISSUER = "https://dev-192131.okta.com/oauth2/default"
+CLIENT_ID = "0oa967awncYBzmmsG4x6"
+CLIENT_SECRET = "FxrXdp5YBUdPdp3wyaE3Z6NMGMBGXGJN9S3UL1CU"
+SCOPES = "openid profile email"
+REDIRECT_URI = "http://localhost:8000/oauth2/callback"
```
Comment (gonac) [inline advisor_api/settings.py]: We cannot have localhost as redirect address

**Review by gonac — CHANGES_REQUESTED** [review]

File: advisor_api/settings.py (hunk 2 above)
Comment (ireland-advisor) [reply advisor_api/settings.py]: Updated

File: advisor_api/settings.py (hunk 3 above)
Comment (ireland-advisor) [reply advisor_api/settings.py]: These variables have been added to the env. file and won't be uploaded to GitHub 
Therefore, whoever is running it please create a .env file and run `source .env` command
plus, all the .idea files have been removed as well

**Comment by ireland-advisor** [comment]
I have updated the info, move the sensitive info to local files. 
Whoever is using it please add to the .env source file

**Review by gonac — APPROVED** [review]

**Pull request merged**